CompactedSyntax/CompactedPDFSyntaxTest.pdf.json

```diff
@@ -0,0 +1,1658 @@
+        },
+        {
+          "Operator": "gs",
+          "Operands": [
+            {
+              "Type": "Name",
+              "Value": ""
+            }
+          ]
+        },
+        {
+          "Operator": "q",
+          "Operands": []
+        },
+        {
+          "Operator": "w",
+          "Operands": [
+            {
+              "Type": "Integer",
+              "Value": 40
+            }
+          ]
+        },
+        {
+          "Operator": "re",
+          "Operands": [
+            {
+              "Type": "Integer",
+              "Value": 75
+            },
+            {
+              "Type": "Integer",
+              "Value": 75
+            },
+            {
+              "Type": "Integer",
+              "Value": 400
+            },
+            {
+              "Type": "Integer",
+              "Value": 400
+            }
+          ]
+        },
+        {
+          "Operator": "W",
+          "Operands": []
+        },
+        {
+          "Operator": "S",
+          "Operands": []
+        },
+        {
+          "Operator": "rg",
+          "Operands": [
+            {
+              "Type": "Integer",
+              "Value": 1
+            },
+            {
+              "Type": "Real",
+              "Value": 0.0
+            },
+            {
+              "Type": "Real",
+              "Value": 0.0
+            }
+          ]
+        },
+        {
+          "Operator": "re",
+          "Operands": [
+            {
+              "Type": "Integer",
+              "Value": 75
+            },
+            {
+              "Type": "Integer",
+              "Value": 75
+            },
+            {
+              "Type": "Integer",
+              "Value": 200
+            },
+            {
+              "Type": "Integer",
+              "Value": 200
+            }
+          ]
+        },
+        {
+          "Operator": "f",
+          "Operands": []
+        },
+        {
+          "Operator": "rg",
+          "Operands": [
+            {
+              "Type": "Integer",
+              "Value": 0
+            },
+            {
+              "Type": "Integer",
+              "Value": 1
+            },
+            {
+              "Type": "Integer",
+              "Value": 0
+            }
+          ]
+        },
+        {
+          "Operator": "re",
+          "Operands": [
+            {
+              "Type": "Integer",
+              "Value": 275
+            },
+            {
+              "Type": "Integer",
+              "Value": 75
+            },
+            {
+              "Type": "Integer",
+              "Value": 200
+            },
+            {
+              "Type": "Integer",
+              "Value": 200
+            }
+          ]
+        },
+        {
+          "Operator": "f",
+          "Operands": []
+        },
+        {
+          "Operator": "rg",
+          "Operands": [
+            {
+              "Type": "Real",
+              "Value": 0.0
```

Comment:
This should be ".0" not "0.0"

Reply:
Can you help me understand why you want to distinguish between .0 and 0.0 after lexical analysis?

Reply:
Mainly because not all lexers and tokenizers are equal. 
And there are nasty things like signed zeros that can and do occur (not for this specific line, but my other comments elsewhere). 
And parsing JSON is also parser-dependent (see https://seriot.ch/projects/parsing_json.html and https://labs.bishopfox.com/tech-blog/an-exploration-of-json-interoperability-vulnerabilities).
So if the value of "Value" was a string rather than a post-processed number then it would avoid all assumptions but still keep the nicely decomposed structure from the PDF as a more friendly JSON (thank you!). (And, yes, slightly more work for each user)

Reply:
Exactly because not all lexers and tokenisers are equal (both in terms of implementation details and technology) a neutral reference is helpful. By representing a real or integer as a string, verification would still rely on the parsing of the particular lexer. In my view 0.0 is the most neutral representation of all zero real numbers such as -.0, -0.0, +.0, +0.00000, -.00000, etc.

https://json-schema.org/understanding-json-schema/reference/numeric.html explicitly discourages representing a number as a string.